tests/lib/rules/jquery-dollar-sign-reference.js

```diff
@@ -231,6 +231,11 @@ ruleTester.run('jquery-dollar-sign-reference', rule, {
     {code: 'var foo = bar || qux;'},
     {code: 'var foo = bar && quz;'},
     {code: 'var foo = bar || (qux && baz);'},
+
+    {code: 'async function foo() { var foo = await aPromise; }', parserOptions: {ecmaVersion: 2017}},
+    {code: 'async function foo() { var foo = await $aPromise; }', parserOptions: {ecmaVersion: 2017}},
+    {code: 'async function foo() { var $foo = await aPromise; }', parserOptions: {ecmaVersion: 2017}},
+    {code: 'async function foo() { var $foo = await $aPromise; }', parserOptions: {ecmaVersion: 2017}},
```

Comment:
Might want to test this for a few assignment expressions and some awaited things that are function calls, just to cover all the other cases.